packages/editor-ui/src/views/SettingsUsageAndPlan.vue

```diff
@@ -1,44 +1,159 @@
 <script lang="ts" setup>
-import { onMounted } from 'vue';
-import { useUsageStore } from '@/stores/usage';
+import { computed, onMounted, ref, watch } from 'vue';
+import { useRoute, useRouter } from 'vue-router/composables';
+import { Notification } from "element-ui";
+import { UsageTelemetry, useUsageStore } from '@/stores/usage';
+import { UsageState } from "@/Interface";
+import { telemetry } from "@/plugins/telemetry";
 
 const usageStore = useUsageStore();
+const route = useRoute();
+const router = useRouter();
 
-onMounted(() => {
-	usageStore.getData();
+const queryParamCallback = ref<string>(`callback=${encodeURIComponent(`${window.location.origin}${window.location.pathname}`)}`);
+const viewPlansUrl = computed(() => `${usageStore.viewPlansUrl}&${queryParamCallback.value}`);
+const managePlanUrl = computed(() => `${usageStore.managePlanUrl}&${queryParamCallback.value}`);
+const activationKeyModal = ref(false);
+const activationKey = ref('');
+const activationKeyInput = ref<HTMLInputElement | null>(null);
+
+const onLicenseActivation = () => {
+	usageStore.activateLicense(activationKey.value).then(() => {
+		activationKeyModal.value = false;
+	});
+};
+
+onMounted(async () => {
+	usageStore.setLoading(true);
+	if(route.query.key) {
+		await usageStore.activateLicense(route.query.key as string).then(() => router.replace({ query: {} }));
+	} else if(usageStore.canUserActivateLicense) {
+		await usageStore.refreshLicenseManagementToken();
+	} else {
+		await usageStore.getLicenseInfo();
+	}
+	usageStore.setLoading(false);
+});
+
+watch(() => usageStore.error, (error: UsageState['error']) => {
+	if(error?.message) {
+		Notification.error({
+			title: 'Error',
+			message: error.message,
+			position: 'bottom-right',
+		});
+	}
 });
 
+watch(() => usageStore.success, (success: UsageState['success']) => {
+	if(success?.title && success?.message) {
```

Comment:
why do we need to store the success state in the store? a simpler pattern that we follow everywhere is..
```
async function activate() {
try {
   await store.activate();
   showMessage('success')
} catch (e){
  showMessage('error);
}
```
no watchers, no anything

Reply:
That pattern is breaking the single responsibility rule. The store should just deal with with the state and not render, the component is meant to render depending on the state.

I really don't like `showMessage` in store but if there is no other way I can move Notification call to the store but then I have to at least add it to every action that makes a network request

Reply:
hmm I don't mean `showMessage` in store.. this should be within the component itself.. so only the component deals with rendering.

```
SettingsUsageAndPlan.vue
async function activate() {
try {
   await store.activate();
   showMessage('success')
} catch (e){
  showMessage('error);
}
```

Reply:
Ok, it will get a bit uglier because I need to use `showMessage` in more places (everywhere where I call a store action) instead of just in the watcher but I'll do it

Reply:
seems to me this is cleaner. especially when you have multiple actions in the same view.. or when you reuse the same action across multiple views.. most stores are not 1:1 with views.. we would need a different approach to notifications across the app to do this

Reply:
to do this I would..
1. add notification stack to ui store or new notifications store. (like modal stack)
2. add a global functions to root that render the notification stack
3. from stores or components, push to the notifications stack different types..

Reply:
Yes, this sounds good but probably I should not start doing it now

Reply:
yes ofc